File: solutions/Leetcode_746/leetcode_746.cc
#include <vector>

using namespace std;

class Solution1
{
public:
    int minCostClimbingStairs(vector<int> &cost)
    {
        int prev1 = 0, prev2 = 0;
        for (int i = 0; i < cost.size(); i++)
        {
            int tmp = prev2;
            prev2 = cost[i] + min(prev1, prev2);
            prev1 = tmp;
        }
        return min(prev1, prev2);
    }
};

File: solutions/Leetcode_746/leetcode_746.py
from typing import List


class Solution1:
    def min_cost_climbing_stairs(self, cost: List[int]) -> int:
        prev1, prev2 = 0, 0
        for i in range(len(cost)):
            prev1, prev2 = prev2, cost[i] + min(prev1, prev2)
        return min(prev1, prev2)
